File: orderstream-backend/src/models/MenuItem.js
import mongoose from 'mongoose';

const menuItemSchema = new mongoose.Schema(
    {
        name: { type: String, required: true },
        description: String,
        category: { type: String, required: true },
        price: { type: Number, required: true },
        imageUrl: String,
        rating: { type: Number, default: 0 },
        available: { type: Boolean, default: true }
    },
    { timestamps: true }
);

export default mongoose.model('MenuItem', menuItemSchema);

File: orderstream-backend-fastapi/app/models/menu_item.py
from sqlalchemy import Column, Integer, String, Float, Boolean, DateTime
from sqlalchemy.sql import func
from app.database import Base

class MenuItem(Base):
    __tablename__ = "menu_items"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    description = Column(String)
    category = Column(String, nullable=False, index=True)
    price = Column(Float, nullable=False)
    image_url = Column(String)
    rating = Column(Float, default=0.0)
    available = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())